include/fi_util.h

```diff
@@ -295,6 +295,32 @@ struct util_event {
 int fi_eq_create(struct fid_fabric *fabric, struct fi_eq_attr *attr,
 		 struct fid_eq **eq_fid, void *context);
 
+/*
+ * Buffer Pool
+ */
+
+struct util_buf_pool {
+	size_t size;
+	size_t num;
+	void *memory_region;
+	struct slist list;
+	fastlock_t lock;
```

Comment:
@shefty - The intend was to provide both lock and lock-free versions. But, makes sense to have only the lock-free version. I will update it.